Tests/Unit/Office365DSC/Office365DSC.SPOSite.Tests.ps1

```diff
@@ -799,6 +799,128 @@ Describe -Name $Global:DscHelper.DescribeHeader -Fixture {
              }
          }
 
+        Context -Name "Group#0 site does not exist" -Fixture {
+            $testParams = @{
+                Url                                      = "https://contoso.sharepoint.com/sites/groupsite"
+                Owner                                    = "testuser@contoso.com"
+                StorageQuota                             = 1000
+                GlobalAdminAccount                       = $GlobalAdminAccount
+                Ensure                                   = "Absent"
+                LocaleId                                 = 1033
+                Template                                 = "GROUP#0"
+                CompatibilityLevel                       = 15
+                Title                                    = "groupsite"
+                DenyAddAndCustomizePages                 = $false
+                StorageQuotaWarningLevel                 = 25574400
+                LockState                                = "Unlock"
+                SharingCapability                        = "Disabled"
+                CommentsOnSitePagesDisabled              = $false
+                SocialBarOnSitePagesDisabled             = $false
+                DisableAppViews                          = "NotDisabled"
+                DisableCompanyWideSharingLinks           = "NotDisabled"
+                DisableFlows                             = "NotDisabled"
+                RestrictedToGeo                          = "BlockMoveOnly"
+                SharingDomainRestrictionMode             = "None"
+                SharingAllowedDomainList                 = ""
+                SharingBlockedDomainList                 = ""
+                ShowPeoplePickerSuggestionsForGuestUsers = $false
+                DefaultSharingLinkType                   = "None"
+                DefaultLinkPermission                    = "None"
+            }
+
+            Mock -CommandName Set-SPOSite -MockWith { }
+
+            Mock -CommandName Get-SPOSite -MockWith {
+                return $null
+            }
+
+            It "Should return absent from the Get method" {
+                (Get-TargetResource @testParams).Ensure | Should Be "absent"
+            }
+
+            It "Should return false from the Test method" {
+                Test-TargetResource @testParams | Should Be $false
+            }
+
+            It "should not create the site and highlight that it should be created via the O365 group" {
+                Set-TargetResource @testParams
```

Comment:
This should throw an error based on the logic. Make sure you do:

Set-TargetResource @testParams | Should Throw "blah"

Take a look at the following for more info:
https://github.com/PowerShell/SharePointDsc/blob/dev/Tests/Unit/SharePointDsc/SharePointDsc.SPAlternateUrl.Tests.ps1#L221